--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -302,13 +302,25 @@ else(ROS_EDITION STREQUAL "ROS2")
   endif()
 
   # include file direcotry
-  target_include_directories(${PROJECT_NAME} PUBLIC
-    ${PCL_INCLUDE_DIRS}
-    ${APR_INCLUDE_DIRS}
-    ${LIVOX_LIDAR_SDK_INCLUDE_DIR}
-    ${LIVOX_INTERFACES_INCLUDE_DIRECTORIES}   # for custom msgs
-    3rdparty
-    src
-  )
+  if(HUMBLE_ROS STREQUAL "humble")
+    target_include_directories(${PROJECT_NAME} PUBLIC
+      ${PCL_INCLUDE_DIRS}
+      ${APR_INCLUDE_DIRS}
+      ${LIVOX_LIDAR_SDK_INCLUDE_DIR}
+      3rdparty
+      src
+    )
+  else()
+    target_include_directories(${PROJECT_NAME} PUBLIC
+      ${PCL_INCLUDE_DIRS}
+      ${APR_INCLUDE_DIRS}
+      ${LIVOX_LIDAR_SDK_INCLUDE_DIR}
+      ${LIVOX_INTERFACES_INCLUDE_DIRECTORIES}
+      3rdparty
+      src
+    )
+  endif()
 
   # link libraries
   target_link_libraries(${PROJECT_NAME}
